src/data.ts

```diff
@@ -240,30 +267,31 @@ export class Data {
 				}) as IProfile);
 			}
 		});
-		profiles.forEach((it) => {
-			db.profiles.put(it);
-		});
-		// db.profiles.bulkPut(profiles)
+
+		queueMicrotask(()=>{
+			db.profiles.bulkPut(profiles)
+		})
+		await yieldMicrotask()
```

Comment:
The reason for the bulkPut is for efficiency, from the Dexie docs:

> If you have a large number of objects to add to the object store, bulkPut() is faster than doing put() in a loop. Mainly because it only has to get called back for the last onsuccess of all indexedDB requests. (https://dexie.org/docs/Table/Table.bulkPut())

As for the microtask, it was unneeded in this case, but its a way to manually enqueue microtasks to run after a function exits if the stack is empty, but before returning control to the event loop